mlflow/server/handlers.py

```diff
@@ -560,6 +575,16 @@ def _search_runs():
 @catch_mlflow_exception
 @_disable_if_artifacts_only
 def _list_artifacts():
+
+    if _get_serve_artifacts_mode():
+        return _list_artifacts_mlflow_artifacts()
```

Comment:
I tested this out manually and discovered that this approach isn't quite correct. `ListArtifactsMlflowArtifacts` interprets the `path` parameter differently than `ListAritfacts`:

- `ListArtifactsMlflowArtifacts` interprets the `path` relative to the root directory of the configured `--artifacts-destination`. For example, supplying `"path": "1"` should list artifacts under `mlflow-artifacts:/1`.

- `ListArtifacts` interprets the `path` relative to the root artifact storage location of the run corresponding to the specified `run_id`, which is a *subdirectory* of the `--artifacts-destination`. Example: For a run with artifact root location `mlflow-artifacts:/0/60e69bf052914090b92f8b7bf1ceb0b9/artifacts`, a `ListArtifacts` request with `"path": "1"` should list artifacts under `mlflow-artifacts:/0/60e69bf052914090b92f8b7bf1ceb0b9/artifacts/1`.

This currently results in recursive listing behavior in the UI where every `ListArtifacts` call returns a list of artifacts from the root of `mlflow-artifacts:/`:

https://user-images.githubusercontent.com/39497902/155090740-2c2ba882-ea1c-4655-b92c-48207b60fa4d.mov


To fix this, we need to fetch the artifact root location of the specified run and forward it to `_list_artifacts_mlflow_artifacts()`.

Reply:
Fix is in. Validated (to be extra certain) with manual docker env checks with a remote fs container, and separate artifact server / tracking server instances. Artifact listing from client and REST API access looks good and returns correct pathing for a submitted runid listing and runid + path listings.